flink-clients/src/test/java/org/apache/flink/client/deployment/application/ClassPathPackagedProgramRetrieverTest.java

```diff
@@ -285,25 +148,55 @@ public void testJobGraphRetrievalFailIfDoesNotFindTheEntryClassInTheJobDir()
     }
 
     @Test
-    public void testRetrieveCorrectUserClasspathsWithoutSpecifiedEntryClass()
+    public void testSavepointRestoreSettings()
```

Comment:
This should also be tested for `JarFilePackagedProgramRetriever` and `PythonBasedPackagedProgramRetriever`. Maybe we could add it to the `PackagedProgramRetrieverTestBase`.